internal/resource/persistence/template.go

```diff
@@ -63,9 +63,11 @@ var (
 		`),
 		SetupAdvancedVisibility: dedent.Dedent(`
 			#!/bin/bash
+			# Change index_patterns from temporal_visibility_v1* to {{ .Indices.Visibility }}* at index_template_{{ .Version }}.json before apply 
+			sed -i '' -e 's/temporal_visibility_v1./{{ .Indices.Visibility }}*/g' /etc/temporal/schema/elasticsearch/visibility/index_template_{{ .Version }}.json
```

Comment:
According to: `https://stackoverflow.com/a/12696224/4965457` the `-i ''` tricks is used for macOS.
```suggestion
			sed -i -e 's/temporal_visibility_v1./{{ .Indices.Visibility }}*/g' /etc/temporal/schema/elasticsearch/visibility/index_template_{{ .Version }}.json
```

Otherwise LGTM.